howto/spark/spark_mlrun_describe.ipynb

```diff
@@ -0,0 +1,1417 @@
+    "    except (ValueError, KeyError):\n",
+    "        pass\n",
+    "\n",
+    "    return table_stats, variable_stats.T, freq_dict"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from pyspark.sql.functions import abs as absou\n",
+    "\n",
+    "SKEWNESS_CUTOFF = 20\n",
+    "DEFAULT_FLOAT_FORMATTER = u'spark_df_profiling.__default_float_formatter'\n",
+    "\n",
+    "\n",
+    "def gradient_format(value, limit1, limit2, c1, c2):\n",
+    "    def LerpColour(c1,c2,t):\n",
+    "        return (int(c1[0]+(c2[0]-c1[0])*t),int(c1[1]+(c2[1]-c1[1])*t),int(c1[2]+(c2[2]-c1[2])*t))\n",
+    "    c = LerpColour(c1, c2, (value-limit1)/(limit2-limit1))\n",
+    "    return fmt_color(value,\"rgb{}\".format(str(c)))\n",
+    "\n",
+    "\n",
+    "def fmt_color(text, color):\n",
+    "    return(u'<span style=\"color:{color}\">{text}</span>'.format(color=color,text=str(text)))\n",
+    "\n",
+    "\n",
+    "def fmt_class(text, cls):\n",
+    "    return(u'<span class=\"{cls}\">{text}</span>'.format(cls=cls,text=str(text)))\n",
+    "\n",
+    "\n",
+    "def fmt_bytesize(num, suffix='B'):\n",
+    "    for unit in ['','Ki','Mi','Gi','Ti','Pi','Ei','Zi']:\n",
+    "        if num < 0:\n",
+    "            num = num*-1\n",
+    "            if num < 1024.0:\n",
+    "                return \"%3.1f %s%s\" % (num, unit, suffix)\n",
+    "            num /= 1024.0\n",
+    "    return \"%.1f %s%s\" % (num, 'Yi', suffix)\n",
+    "\n",
+    "\n",
+    "def fmt_percent(v):\n",
+    "    return  \"{:2.1f}%\".format(v*100)\n",
+    "\n",
+    "def fmt_varname(v):\n",
+    "    return u'<code>{0}</code>'.format(v)\n",
+    "\n",
+    "\n",
+    "value_formatters={\n",
+    "        u'freq': (lambda v: gradient_format(v, 0, 62000, (30, 198, 244), (99, 200, 72))),\n",
+    "        u'p_missing': fmt_percent,\n",
+    "        u'p_infinite': fmt_percent,\n",
+    "        u'p_unique': fmt_percent,\n",
+    "        u'p_zeros': fmt_percent,\n",
+    "        u'memorysize': fmt_bytesize,\n",
+    "        u'total_missing': fmt_percent,\n",
+    "        DEFAULT_FLOAT_FORMATTER: lambda v: str(float('{:.5g}'.format(v))).rstrip('0').rstrip('.'),\n",
+    "        u'correlation_var': lambda v: fmt_varname(v),\n",
+    "        u'unparsed_json_types': lambda v: ', '.join([s.__name__ for s in v])\n",
+    "        }\n",
+    "\n",
+    "def fmt_row_severity(v):\n",
+    "    if np.isnan(v) or v<= 0.01:\n",
+    "        return \"ignore\"\n",
+    "    else:\n",
+    "        return \"alert\"\n",
+    "\n",
+    "def fmt_skewness(v):\n",
+    "    if not np.isnan(v) and (v<-SKEWNESS_CUTOFF or v> SKEWNESS_CUTOFF):\n",
+    "        return \"alert\"\n",
+    "    else:\n",
+    "        return \"\"\n",
+    "\n",
+    "row_formatters={\n",
```

Comment:
There is some "main" code through the cells, could you please write a comment line to describe them?